docs/developer/architecture/authentication.rst

```diff
@@ -0,0 +1,88 @@
+Authentication Architecture Document
+====================================
+
+Context:
+--------
+Currently, when we deploy MetalK8s we pre provision super admin user with admin
+password. Meaning anyone wanting to use K8S/Salt APIs should authenticate using
+this single super admin user.
+This is not good enough in term of security and flexibility.
+
+We would like at least to be able to add different users with different
```

Comment:
As said above, you *can* do that today (though there are security aspects in the current solution that can't be ignored).

Reply:
updated